### PR TITLE
tracker(dm): remove NO_AUTO_CREATE_USER from sql_mode used by dm

### LINKED ISSUE #8149: DM uses a SQL mode which is removed in MySQL 8.0
### What did you do?

https://asktug.com/t/topic/1000577/6

```
[2023/01/31 11:40:41.469 +08:00] [ERROR] [baseconn.go:184] [“execute statement failed”] [task=alydm] [unit=“binlog replication”] [query=“SET SESSION SQL_MODE = ‘ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION’”] [argument=”[]”] [error=“Error 1231: Variable ‘sql_mode’ can’t be set to the value of ‘NO_AUTO_CREATE_USER’”]
```

https://github.com/pingcap/tiflow/blob/b633cef010a52a152c3ab6d2bd27cb1da563d365/dm/pkg/schema/tracker.go#L477-L478

Although we don't support MySQL 8.0 as downstream, this is easy to fix so maybe we should consider it.

### What did you expect to see?

_No response_

### What did you see instead?

_No response_

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by liumengya94**
### What problem does this PR solve?

Issue Number: close #8149

### What is changed and how it works?
`NO_AUTO_CREATE_USER` is removed in MySQL8.0([doc](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-11.html#mysqld-8-0-11-deprecation-removal))，so it should not be used in dm.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/pkg/schema/tracker.go [hunk 1]
```diff
@@ -89,6 +89,8 @@ type executorContext struct {
 
 var _ sqlexec.RestrictedSQLExecutor = executorContext{}
 
+const DefaultSQLMode = "ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
```
Comment (lance6716) [inline dm/pkg/schema/tracker.go:92]:
```suggestion
const DefaultSQLModeMySQL80 = "ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dm/pkg/schema/tracker.go (hunk 1 above)
Comment (D3Hunter) [reply dm/pkg/schema/tracker.go]: why add suffix 80, seems we didn't check mysql version and used it directly.

Comment (lance6716) [reply dm/pkg/schema/tracker.go]: in case other developers "correct" it to MySQL v5

**Pull request closed without merging**